ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/EpochProcessorUtil.java

```diff
@@ -99,23 +101,13 @@ public static UnsignedLong get_previous_total_balance(BeaconState state) {
   public static List<Integer> get_attesting_indices(
       BeaconState state, List<PendingAttestation> attestations) throws IllegalArgumentException {
 
-    List<ArrayList<Integer>> validator_index_sets = new ArrayList<>();
-
-    for (PendingAttestation attestation : attestations) {
-      validator_index_sets.add(
-          get_attestation_participants(
-              state, attestation.getData(), attestation.getAggregation_bitfield()));
-    }
-
-    List<Integer> attester_indices = new ArrayList<>();
-    for (List<Integer> validator_index_set : validator_index_sets) {
-      for (Integer validator_index : validator_index_set) {
-        if (!attester_indices.contains(validator_index)) {
-          attester_indices.add(validator_index);
-        }
-      }
+    HashSet<Integer> output = new HashSet<>();
+    for (PendingAttestation a : attestations) {
+      output.addAll(get_attestation_participants(state, a.getData(), a.getAggregation_bitfield()));
```

Comment:
Just to check: are you 100% that addAll doesn't create duplicates in the hashset here?

Reply:
Yes, HashSets do not allow duplicates.